### DISCUSSION TITLE
Who's using Uno in production?

### DISCUSSION THREAD

**Discussion by aronmek** [unoplatform/uno#5012] [General]
Can you list some names of real companies/apps built on the uno platform?
Also the history of the project wouldn't hurt :)

**Comment by SirAnimesh** [comment]
This *.NET Rocks* [podcast episode](https://pca.st/06Og) covers Uno in depth. Hope it helps.

**Comment by jeromelaban** [comment]
We're still working out the details of sharing details, but here are some apps that have been published:
- https://itunes.apple.com/ca/app/jean-coutu/id351461407
- https://itunes.apple.com/ca/app/my-md-mobile/id1144752656
- https://itunes.apple.com/us/app/vca-careclub/id1172429469

**Comment by ghuntley** [comment]
Howdy folks,

As @jeromelaban has alluded to Uno has been extensively used by https://nventive.com/ to ship production applications to their customers. There's some they can talk about and a whole heap they cannot. 

If you are using Uno in production, we would love to hear about it. Please leave a comment below with where you are using Uno, what problems it solved for you and the platforms used. 

# template
```
<share a screenshot or two by dragging it onto the comment>

Author: 
Website: 

Apple Store: 
Android Store: 
Windows Store:
WebAssembly Url: 

If open-source, where is the git repo?:
```

**Comment by TopperDEL** [comment]
I'm using it for a - german - app to help my theater-group organize our Performances and Trainings. The app is open for everyone and Shows our next Performances. It has a secret area where only we can get into and work with it.
It currently is in beta - there are many todos left. :)
![Screenshot_20190523_202709_de vollemoehre mobilmoehre](https://user-images.githubusercontent.com/1833242/58277083-37bb4980-7d99-11e9-98ad-00eb78369474.jpg)
![Screenshot_20190523_202657_de vollemoehre mobilmoehre](https://user-images.githubusercontent.com/1833242/58277097-3db12a80-7d99-11e9-9434-51eee21eebc9.jpg)

Author: Tim Parth
Website: www.vollemoehre.de

Apple Store: not yet available
Android Store: [Store-Entry](https://play.google.com/store/apps/details?id=de.vollemoehre.mobilmoehre&hl=de)
Windows Store: not yet available
WebAssembly Url: not yet available

Unfortunately closed source, as this app is mostly for internal use.

I'm also in development of another app that will be open source but Comes from a private closed-source Project. I Need more time until I can make it public. It will be kind of a Diary app. More to come on this place once it's Ready (enough :) ).

**Comment by ghuntley** [comment]
Here's a list of live apps made with the Uno Platform for WebAssembly.

* The [Uno Platform Playground](https://playground.platform.uno) ([Source](https://github.com/nventive/Uno.Playground))
* The [Xaml Controls Gallery](https://xamlcontrolsgallery.platform.uno/) ([Source](https://github.com/nventive/Uno.Xaml-Controls-Gallery))
* The [Uno.WindowsCommunityToolkit](https://windowstoolkit-wasm.platform.uno/), ([Source](https://github.com/nventive/Uno.WindowsCommunityToolkit))
* The [Uno.Lottie](https://lottie.platform.uno/), a sample that uses the [AnimatedVisualPlayer](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.animatedvisualplayer) ([Source](https://github.com/nventive/Uno.LottieSample))
* The [Uno.RoslynQuoter](http://roslynquoter-wasm.platform.uno/), a [Roslyn](https://github.com/dotnet/roslyn) based C# analysis tool ([Source](https://github.com/nventive/Uno.RoslynQuoter))
* The [Uno.BikeSharing360 App](http://bikerider-wasm.platform.uno/), a Xamarin.Forms app running on top of Uno for WebAssembly ([Source](https://github.com/nventive/Uno.BikeSharing360_MobileApps))
* The [Uno.WindowsStateTriggers App](http://winstatetriggers-wasm.platform.uno/), a demo of the [Morten's WindowsStateTriggers](https://github.com/dotMorten/WindowsStateTriggers) ([Source](https://github.com/nventive/Uno.WindowsStateTriggers))
* The [SQLite + Entity Framework Core App](http://sqliteefcore-wasm.platform.uno), a demo of the combination of [Roslyn](https://github.com/dotnet/roslyn), [Entity Framework Core](https://docs.microsoft.com/en-us/ef/core/), [SQLite](https://github.com/nventive/Uno.SQLitePCLRaw.Wasm) and the Uno Platform to manipulate an in-browser database.
* The [Uno.WebSockets App](https://websockets-wasm.platform.uno), a demo of System.Net.WebSocket running from WebAssembly ([Source](https://github.com/nventive/Uno.Wasm.WebSockets))
* A [mono-wasm AOT RayTracer](https://raytracer-mono-aot.platform.uno/)

**Comment by ghuntley** [comment]
Stumbled upon this tweet by @gritsenko.

https://twitter.com/Pix2d/status/1153326997029949441?s=09

Pix2D, an application for developing pixel art for games uses Uno!

![image](https://user-images.githubusercontent.com/127353/61765335-0229ef80-ae20-11e9-9dae-66543813d686.png)


https://play.google.com/store/apps/details?id=com.pix2d.pix2dapp&hl=en



<blockquote><img src="https://pbs.twimg.com/profile_images/826023502478184448/DisL8fSS_400x400.jpg" width="48" align="right"><div>Twitter</div><div><strong><a href="https://twitter.com/Pix2d/status/1153326997029949441">Igor Gritsenko on Twitter</a></strong></div><div>“Hey #gamedev and #pixelart fellows! #Pix2d now available in Google Play for public beta. Need your feedback to keep myself to be productive! What features should be added first? 
Made with #uwp @UnoPlatform, @xamarinhq and #skiasharp 
https://t.co/oYVa8jDeNt”</div></blockquote>

**Comment by TopperDEL** [comment]
I did a sample-app for another open source component called "uplink.NET" (see [here](https://github.com/TopperDEL/uplink.net)). The library helps connecting to the storj-network ([storj.io](https://storj.io)) a private, secure and decentralized storage network.

The sample app shows how to use the library to upload and download files. It currently works on UWP and Android. iOs is not supported because uplink.NET does not have the necessary binaries.

![uplink1](https://user-images.githubusercontent.com/1833242/65378729-fe4e0880-dcbc-11e9-9c96-7e7aee2f1893.jpg)
![uplink3](https://user-images.githubusercontent.com/1833242/65378730-0148f900-dcbd-11e9-80dd-795778623593.jpg)
![uplink2](https://user-images.githubusercontent.com/1833242/65378732-06a64380-dcbd-11e9-99b2-7b7885121cd7.jpg)
![uplinkVideo](https://user-images.githubusercontent.com/1833242/65378802-08bcd200-dcbe-11e9-9364-8b608520a610.gif)

**Comment by MartinZikmund** [comment]
I have brought my old Windows Phone app for purchasing SMS tickets to Android and iOS via Uno Platform. It is doing rather well, especially on Android where it has already over 3000 downloads and counting :-) . It uses some of the new APIs like SMS compose and JumpLists and is localized into Czech and English. 

iOS App Store: https://apps.apple.com/us/app/czech-sms-ticket/id1477710726
Android Play Store: https://play.google.com/store/apps/details?id=com.sphereline.smsticketandroid


![EDeq6J2W4AUZpic](https://user-images.githubusercontent.com/1075116/65903157-37bcfd00-e3bc-11e9-8d3c-8817dc400e57.jpg)
![EDeq6eyWwAEG-Pg](https://user-images.githubusercontent.com/1075116/65903171-3ab7ed80-e3bc-11e9-8ddd-d2b849274b6c.jpg)
![EDeq71dXoAAHXTe](https://user-images.githubusercontent.com/1075116/65903177-3d1a4780-e3bc-11e9-8813-21dc69dadb06.jpg)



<blockquote><img src="https://is5-ssl.mzstatic.com/image/thumb/Purple113/v4/e7/53/03/e7530360-ae53-76a4-371c-f25adf652ee5/AppIcons-0-0-1x_U007emarketing-0-0-0-7-0-85-220.png/1200x630wa.png" width="48" align="right"><div>App Store</div><div><strong><a href="https://apps.apple.com/us/app/czech-sms-ticket/id1477710726">‎Czech SMS Ticket</a></strong></div><div>‎This application allows you to quickly purchase a ticket for Czech public transportation with a SMS. Supported cities are:

- Prague
- Brno
- Carlsbad
- Liberec
- České Budějovice
- Hradec Králové
- Olomouc
- Ostrava
- Pardubice
- Pilsen
- Sokolov
- Tábor
- Ústí nad Labem
- Zlín and Otrokovice

You…</div></blockquote>
<blockquote><img src="https://lh3.googleusercontent.com/jVD8ZrULoWLiecI7YcimQkaxyI6AnCkg6bXSPfFIODvTiTG1x8Bfyai6enX0mN8PBw" width="48" align="right"><div><strong><a href="https://play.google.com/store/apps/details?id=com.sphereline.smsticketandroid&hl=en_US">Czech SMS Ticket - Apps on Google Play</a></strong></div><div>This application allows you to quickly purchase a ticket for Czech public transportation with a SMS. Supported cities are:

- Prague
- Brno
- Karlovy Vary
- Liberec
- České Budějovice
- Hradec Králové
- Olomouc
- Ostrava
- Pardubice
- Pilsen
- Sokolov
- Ústí nad Labem
- Tábor
- Zlín and Otrokovice</div></blockquote>

**Comment by mizzleinetimi** [comment]
i believe accunews makes use of it
https://www.accunewsapp.com/

**Comment by ghuntley** [comment]
The Monkey Fest conference website is built with Uno's Xamarin Forms WebAssembly custom renderers.

https://javiersuarezruiz.wordpress.com/2019/10/16/xamarin-forms-en-la-web-con-webassembly-y-uno/

Source code is available at https://github.com/jsuarezruiz/monkeyconf-website

<blockquote><img src="https://javiersuarezruiz.files.wordpress.com/2019/10/wasm-project.png" width="48" align="right"><div><img src="https://s1.wp.com/i/favicon.ico" height="14"> Javier Suárez | Blog</div><div><strong><a href="https://javiersuarezruiz.wordpress.com/2019/10/16/xamarin-forms-en-la-web-con-webassembly-y-uno/">Xamarin.Forms en la web con WebAssembly y UNO</a></strong></div><div>Introducción Xamarin.Forms en la web. No es la primera vez que hablamos de ello. Dependiendo de con quien hables es una idea que trae malos recuerdos y piensa en algo horrible o piensa que es genia…</div></blockquote>
<blockquote><img src="https://avatars0.githubusercontent.com/u/6755973?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/jsuarezruiz/monkeyconf-website">jsuarezruiz/monkeyconf-website</a></strong></div><div>Website project of the Monkey Conf event created using Xamarin.Forms. - jsuarezruiz/monkeyconf-website</div></blockquote>

**Comment by pkar70** [comment]
And my app, simple "merge" of Wikipedia 'today' pages from languages... As it seems that very different birthdays, deaths and events are celebrated in different language regions. In effect, you get more events  than in any single language.

https://play.google.com/store/apps/details?id=pkar.Anniversaries 
https://www.amazon.com/dp/B087JVLLVG/
(only for Android, as I have no access to iOS to test).

UWP version: https://www.microsoft.com/en-us/p/a/9pjw858v28w8

<blockquote><img src="https://lh3.googleusercontent.com/x-DS1_Y1iddddlmYRDSxIPd97CSNstvVRHsPzCmq9u48ttJYG4lbWrZZ5cV9DAavjqr0" width="48" align="right"><div><strong><a href="https://play.google.com/store/apps/details?id=pkar.Anniversaries&hl=en_US">Anniversaries - Apps on Google Play</a></strong></div><div>This simple app merges information from "today" Wikipedia pages for some languages. In this way you get two benefits: 
1) many more entries that in any single language;
2) you can see which events are important for which language group;

E.g., you can check who's birthday we celebrate at same time as you celebrate yours? And not only in yours language circle, but also in other regions.

As app presents large amount of text, big screen size is recommended.

Primary version of this app is for Windows (with same name). Migration to Android is done using Platform Uno.</div></blockquote>



<blockquote><img src="//store-images.s-microsoft.com/image/apps.45931.14278620089627205.bf5e269c-cd51-43ee-ba26-5d5ed808b5a3.85b3c164-16a2-4056-8d8b-24b885ce28bc?w=120&h=120&q=60" width="48" align="right"><div>Microsoft Store</div><div><strong><a href="https://www.microsoft.com/en-us/p/anniversaries/9pjw858v28w8">Get ANNIVERSARIES - Microsoft Store</a></strong></div><div>Download this app from Microsoft Store for Windows 10, Windows 10 Mobile, Windows 10 Team (Surface Hub), HoloLens, Xbox One. See screenshots, read the latest customer reviews, and compare ratings for ANNIVERSARIES. </div></blockquote>

**Comment by pkar70** [comment]
And my second app, simple "self-test" of character's dynamism. For two years, it was only UWP and only Polish; as I could not find where I can set default lang = "pl" in Uno, I added English version...

https://play.google.com/store/apps/details?id=pkar.MazurCiC
https://www.amazon.com/dp/B087NXN38F/

(only for Android, as I have no access to iOS to test).
UWP version: https://www.microsoft.com/en-us/p/d/9npj2m5d8lf4
UWP has one feature not present in Android version - prognosis of relation (e.g. marriage).

<blockquote><img src="https://lh3.googleusercontent.com/JhqnNy2otFuCbzkYIxHDUxi_5LCSDzkmhyQXXAldQMauU4NST8X4pgGCYgQRHtKK-HU" width="48" align="right"><div><strong><a href="https://play.google.com/store/apps/details?id=pkar.MazurCiC&hl=en_US">Character dynamism - Apps on Google Play</a></strong></div><div>This app allows you to check dynamism of your character, and also allows to make some prognosis about how it change, as dynamism of every person is constantly changing, from exodynamic (at birth) to statism to endodynamic.


This self-test is based on "Cybernetics and character" book of polish cybernetician, prof. Marian Mazur. 

App presents long texts, so it is better to use it on desktop or tablet.

Base version is for Windows. Porting to Android is done by using Platform Uno.
Windows version (same name) has additionally comparing of characters of two person, with prognosis about their relation.</div></blockquote>
<blockquote><img src="//store-images.s-microsoft.com/image/apps.62566.13788439394442517.9a6a0761-989c-4c12-aff4-6800faaa5f4e.35011b19-8139-419f-bd94-efa9de5fdbbc?w=120&h=120&q=60" width="48" align="right"><div>Microsoft Store</div><div><strong><a href="https://www.microsoft.com/en-us/p/character-dynamism/9npj2m5d8lf4">Get Character dynamism - Microsoft Store</a></strong></div><div>Download this app from Microsoft Store for Windows 10, Windows 10 Mobile, Windows 10 Team (Surface Hub), HoloLens, Xbox One. See screenshots, read the latest customer reviews, and compare ratings for Character dynamism. </div></blockquote>

**Comment by pkar70** [comment]
My third app, public transport in Kraków (Poland).
 Android version is stripped-down version of UWP - e.g. MapControl doesnt work (so you cannot choose location from map), files in package cannot be referenced (so no history of tram network), no Grid.ContextFlyout (going to tram/bus route, and sorting, is unavailable).

https://play.google.com/store/apps/details?id=pkar.KrakTram

UWP version: https://www.microsoft.com/en-us/p/k/9nc74c3g8kgn

<blockquote><img src="https://lh3.googleusercontent.com/TwKNVI3z-H9x4uZebImOn88Szu9B97p0Kgw1cwLkHQ3EVkrrXufxFfnRcfXw1HJWwHQ" width="48" align="right"><div><strong><a href="https://play.google.com/store/apps/details?id=pkar.KrakTram&hl=en_US">KrakTram - Apps on Google Play</a></strong></div><div>This app shows current, realtime data about when to expect tram and bus in Kraków (Cracow, Poland). It uses your GPS position to show only relevant data (from nearest stops). It is not scheduled information, but real data (with all delays included).
You can sort data by line, direction or stop.

You can also see current reroutes.

This app is ported from UWP (Windows) version, using Platform Uno. UWP version has some feature missing in Android version: history of tram lines (since 1882), context menus on departure page (to change sort mode; you can change sort mode from bottom app bar), choosing location from map.
Android compilation possible thanks to Platform Uno.</div></blockquote>
<blockquote><img src="https://store-images.s-microsoft.com/image/apps.46630.13569232299979803.c3198716-2b9e-4f72-9ff4-8b2acf434f44.3533553a-4526-40df-b91d-842954c570d8%3Fw=120&h=120&q=60" width="48" align="right"><div>Microsoft Store</div><div><strong><a href="https://www.microsoft.com/en-us/p/kraktram/9nc74c3g8kgn">Get KrakTram - Microsoft Store</a></strong></div><div>Download this app from Microsoft Store for Windows 10, Windows 10 Mobile. See screenshots, read the latest customer reviews, and compare ratings for KrakTram. </div></blockquote>

**Comment by pkar70** [comment]
My next app, current state of environment.
Android version is stripped-down version of UWP - e.g. MapControl doesnt work (so you cannot choose location from map), no tiles, no notification - because no timer background execuction, and no data exchange (no RemoteSystem).

https://play.google.com/store/apps/details?id=pkar.EnviroStatus

UWP version: https://www.microsoft.com/en-us/p/s/9PHHVVCP239D

<blockquote><img src="https://lh3.googleusercontent.com/a62hav8jlUGObPTch7vgZ34FYxApf8m63FcNzVtipwCFyYZ0iFSTZZcYHzucqDurnmY" width="48" align="right"><div><strong><a href="https://play.google.com/store/apps/details?id=pkar.EnviroStatus&hl=en_US">SmogMeter - Apps on Google Play</a></strong></div><div>It collects (merges) data from different sources:
* world:
* radioactiveathome
* DarkSky: UV index, cloud cover, visibility...
* Foreca: max/min world temperature 
* NOAA solar wind, magnetic activity, and alerts
* seismicportal.eu: earthquakes
* Europe:
* Airly (Europe)
* European Environmental Agency
* Poland:
* polish government's environment service (Poland)

See also help within app (much more detailed than this info).

This app is available for Windows 10 since February 2019; now it is made universal using Platform Uno. Some features are not available for Android (yet), e.g. choosing point using map; communication with other apps; Live Tiles.
Google policy forbids this app for users age below 13 years.</div></blockquote>
<blockquote><img src="https://store-images.s-microsoft.com/image/apps.59198.14235399480949345.47e7d31a-75b7-42e5-94b8-897b7fdcab62.caf118e0-22f6-41c6-9b00-9afb1237f9f6%3Fw=120&h=120&q=60" width="48" align="right"><div>Microsoft Store</div><div><strong><a href="https://www.microsoft.com/en-us/p/smogmeter/9phhvvcp239d">Get SmogMeter - Microsoft Store</a></strong></div><div>Download this app from Microsoft Store for Windows 10, Windows 10 Mobile, Windows 10 Team (Surface Hub), HoloLens, Xbox One. See screenshots, read the latest customer reviews, and compare ratings for SmogMeter. </div></blockquote>

**Comment by pkar70** [comment]
My next app, for fans of Windows Phone, and forced to use Android for some apps :)

App sends Android notification (toast in UWP parlance) to OneDrive, so it can be read on any device (or special UWP app).

Sister Android:
https://play.google.com/store/apps/details?id=pkar.Andro2UWP

Sister UWP:
https://www.microsoft.com/en-us/p/a/9n9jl02ndn4p



<blockquote><img src="https://lh3.googleusercontent.com/kHPuLfXe8YU9xKU_FWiVmnJqpeyxO0cXJMglqZBVZmoAb5w5XrmK6v0cnYhe-OUMsPbQ" width="48" align="right"><div><strong><a href="https://play.google.com/store/apps/details?id=pkar.Andro2UWP&hl=en_US">Andro2UWP - Apps on Google Play</a></strong></div><div>This app sends notifications from apps from your Android device to Windows device. It is somehow similar to "Your phone" application from Microsoft, but also have some differences:
1. It can send notification also from tablets and from Android Go edition (Your phone cannot be installed on such devices)
2. You can add filters (so not every notification is sent to Windows)
3. Notification are sent to your OneDrive, and can be read on any device (companion app is for any UWP device, e.g. Lumia phone; but it can be read even without any app from any device with WWW browser connected to Internet )
4. Notifications from more than one Android device could be send to Windows device
5. And notification from Android devices can be read on many Windows devices.

But, there is no ability (yet?) to respond to notification via Windows device.

App creates log file (for debug purposes - if you encounter some problems, you can send me this log) locally, but all other files stores on OneDrive (Apps/Andro2UWP folder). These files are:
* notifications (or, in UWP parlance, toasts) - files *.toast.txt, created by Android app;
* sender.renames.txt, created by UWP companion app;
* toasts.filters.txt, created by UWP companion app;

Format of sender.renames.txt:
original.sender.name|renamed.sender

E.g.:
com.whatsapp|WhatsApp
com.bigajwcislo.ostrzegator|o!strzegator


Format of toast.filters.txt:
sender|title|text
you can prefix (or sufix) text/title with '*' (mask character).

E.g.:
com.microsoft.skydrive|Preparing...|
android|* is using battery|
com.google.android.music|Downloading…|
com.whatsapp||WhatsApp Web is currently active</div></blockquote>

**Comment by TopperDEL** [comment]
I've created "NutzMich", a German-App to rent and exchange children toys with other families of my home-town. It uses kind of a "web-of-trust" so that only know members can join the group - but the basic catalog can be browsed by anybody.
The app is open source and free for all. It uses [Storj](https://storj.io) as the backend so does not need any additional infrastructure - even for a simple chat. It get's deployed fully automatically via Azure DevOps to Google Play and the Apple Store.

Author: Tim Parth
Website: [NutzMich.de](https://www.nutzmich.de)
Apple Store: [NutzMich for iOs](https://apps.apple.com/us/app/nutzmich/id1545146982)
Android Store: [NutzMich for Android](https://play.google.com/store/apps/details?id=de.nutzmich.app)

![NutzMichScreenshot1](https://user-images.githubusercontent.com/1833242/104424923-4d8a0d80-5580-11eb-87f8-5c6b4ff504ee.png)
![NutzMichScreenshot2](https://user-images.githubusercontent.com/1833242/104424936-5084fe00-5580-11eb-8b70-3bf8313a3474.png)
